### PR TITLE
Cherry-pick doc fixes onto `docmain`

### PR DESCRIPTION
**Pull request by CalMacCQ**
# Description

Cherry-picking the fixes from #1258 and #1261 onto `docmain` so we don't have to wait for the next release for the fixes to be in the API docs.

# Checklist

- [x] I have performed a self-review of my code.
- [ ] I have commented hard-to-understand parts of my code.
- [ ] I have made corresponding changes to the public API documentation.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] I have updated the changelog with any user-facing changes.

### CODE REVIEW

File: .github/workflows/build-without-conan.yml [hunk 1]
```diff
@@ -85,9 +85,6 @@ jobs:
     - name: Install rapidcheck
       run: |
         cd ${TMP_DIR}
-        wget https://github.com/emil-e/rapidcheck/archive/1c91f40e64d87869250cfb610376c629307bf77d.zip
```
Comment (CalMacCQ) [inline .github/workflows/build-without-conan.yml:88]: Hmm I don't think the changes to these to `.yml` files should be here.

File: pytket/conanfile.py [hunk 2]
```diff
@@ -32,7 +32,7 @@ def package(self):
         cmake.install()
 
     def requirements(self):
-        self.requires("tket/1.2.90@tket/stable")
```
Comment (CalMacCQ) [inline pytket/conanfile.py:35]: This change is becasue I changed a link in a code comment which apparently still requrires bumping the version

Comment (CalMacCQ) [inline pytket/conanfile.py:35]: This change is required because I updated a link in a comment in the source code. Apparently still need a version bump.

**Comment by cqc-alec** [comment]
I don't understand why the "Check tket version bump" step is running, since it is only supposed to run for PRs into `develop`: https://github.com/CQCL/tket/blob/cef4c069383e92c5a03c2642a2536cd0724c1ebb/.github/workflows/build_and_test.yml#L66 .

In any case I think you can just omit the change to the hpp file. It won't affect the generated docs at all. Then you shouldn't need the version bump.

**Comment by cqc-alec** [comment]
Actually, why is this workflow running at all? It should only be run on PRs to `main` and `develop`.

**Comment by CalMacCQ** [comment]
> Actually, why is this workflow running at all? It should only be run on PRs to `main` and `develop`.

I mistakenly made the PR wanting to merge into develop, then switched in the UI. Thats probably why.

**Comment by cqc-alec** [comment]
> Actually, why is this workflow running at all? It should only be run on PRs to `main` and `develop`.

Ah, looks like you originally had the PR going into `develop` which is why the workflow is running. In which case we can just ignore it.

**Pull request merged**